job/job.go

```diff
@@ -128,6 +208,18 @@ func (*jobService) CreateJob(ctx context.Context, job *jobTypes.Job, user *authT
 		return err
 	}
 
+	// use deploy-agent to push the image to tsuru's registry
+	newImageDst, err := builderDeploy(ctx, job, builder.BuildOpts{
+		ImageID: job.Spec.Container.Image,
+	})
+	// we don't want to fail the job creation if the image push fails
+	if err == nil {
+		// deploy the job using the new pushed image
+		job.Spec.Container.Image = newImageDst
```

Comment:
@wpjunior store this outside of Spec, like
job.Status.Image or job.Status.TsuruImage or job.Status.PushedImage

Reply:
I forgot to open the PR as draft, still making changes

Reply:
Also, it makes more sense to store the original image outside of spec and keep the deploy-agent image inside spec

Reply:
kubernetes stores human data on Spec, them processed data on Status, I enjoy this convention

Reply:
Actually, we pass JobSpec to the provisioner and honor it, k8s takes the JobSpec specified and creates the Cronjob object, it's certainly not usual nor is it necessary to have a divergent Spec and Status, there's no need for it. If we want to "save" the original image source it's fine, but we should keep the image being used in JobSpec